test/procore/client_test.rb

```diff
@@ -71,7 +71,7 @@ def test_client_no_token
       store: store,
     )
 
-    assert_raises(Procore::OAuthError) do
+    assert_raises(Procore::APIConnectionError) do
```

Comment:
I thought this strange too... 🤔

Reply:
I _think_ it's getting caught here https://github.com/procore/ruby-sdk/blob/master/lib/procore/requestable.rb#L172

Reply:
Oh right that makes sense since it's now being rescued. Okay, well we should definitely not raise `APIConnectionError` when you're missing a token. 

Can you add another error class for missing token? I don't want to mix missing token with refresh failure

Reply:
👍